src/Java.Interop/Java.Interop/JavaArray.cs

```diff
@@ -362,7 +364,7 @@ public void Dispose ()
 		}
 	}
 	
-	public abstract class JavaPrimitiveArray<T> : JavaArray<T> {
+	public abstract class JavaPrimitiveArray<[DynamicallyAccessedMembers (ConstructorsAndInterfaces)] T> : JavaArray<T> {
```

Comment:
To prevent lines from looking like line noise, we should start requiring that *all* custom attributes take *at least* two lines: one (or more) for the custom attribute(s), and one for the *thing* being attributed.

The downside with that idea is the location of `<` and `>` for generics, and I'm just gonna decree:

 1. `<` is on the line of the containing declaration
 2. *Contents* should be indented an additional tab stop
 3. `>` is on a new line vertically aligned with the start of the declaration.

And we should start consistently having `{` begin on a newline for class declarations.  (We frequently did this *anyway*, just not consistently.)

Thus:

```csharp
public abstract class JavaPrimitiveArray<
        [DynamicallyAccessedMembers (ConstructorsAndInterfaces)] // two tab stops
        T   // two tab stops
> // vertically aligned with start of declaration
    : JavaArray<T> // one tab stop
{
    // … one tab stop
}
```

If we assume/imagine that `JavaArray<T>.ArrayCreator` needed custom attributes, I'd want:

```csharp
internal delegate TArray ArrayCreator<
        [DynamicallyAccessedMembers (…)] // 2 tab stops
        TArray
> (
        ref JniObjectReference reference,  // 2 tab stops
        JniObjectReferenceOptions transfer)
    where TArray : JavaArray<T>; // 1 tab stop
```

The `> (`, while fugly, provides a visual barrier between the end of the type parameters and the beginning of the parameter list.

This only needs to happen when custom attributes are used on type parameters.  If custom attributes aren't required, this is quite compact and readable:

```csharp
internal TArray CreateArray<TArray> (
        ref JniObjectReference reference, // 2 tab stops
        JniObjectReferenceOptions transfer) // 2 tab stops
    where TArray : JavaArray<T> // 1 tab stop
{
    // … 1 tab stop
}
```

Reply:
Ok, I think I've found & reformatted these: used VS's regex search to find them all.

Reply:
I think we found a deficiency in VS's regex search: it doesn't appear to have found or fully fixed the `.tt` files, meaning if (when) we regenerate the `.cs` files form the `.tt` templates, changes are lost.

I think we've fixed all "mismatches" now.

We should probably consider a pre-build step which *explicitly* regenerates the `.cs` files from the `.tt` files, to ensure/require that the templates are the *real* source of truth.  (Though I'm not quite sure how to make this work; the "obvious" route would be to use Target Inputs & Outputs, but if the `.cs` is manually changed w/o a corresponding `.tt` change, the Inputs & Outputs won't cause regeneration…)